meson.build

```diff
@@ -455,7 +455,10 @@ foreach decl : [['IFLA_INET6_ADDR_GEN_MODE',         'linux/if_link.h'],
                 ['IFLA_MACVLAN_FLAGS',               'linux/if_link.h'],
                 ['IFLA_IPVLAN_FLAGS',                'linux/if_link.h'],
                 ['IFLA_PHYS_PORT_ID',                'linux/if_link.h'],
+                ['IFLA_BOND_AD_ACTOR_SYSTEM',        'linux/if_link.h'],
+                ['IFLA_BOND_AD_ACTOR_SYS_PRIO',      'linux/if_link.h'],
                 ['IFLA_BOND_AD_INFO',                'linux/if_link.h'],
+                ['IFLA_BOND_AD_USER_PORT_KEY ',      'linux/if_link.h'],
```

Comment:
It seems not necessary to set all value here. Please only add `IFLA_BOND_AD_ACTOR_SYSTEM` here and drop `IFLA_BOND_AD_INFO`. Then update `missing.h`.

Reply:
Checking in `IFLA_BOND_AD_INFO` in meson.build is for missing.h.
So, at least, `# if !HAVE_IFLA_BOND_AD_INFO` line in missing.h needs to be updated.
If it is updated, then `IFLA_BOND_AD_INFO` checking is not necessary.

But, but,... I noticed that the corresponding block may need to be completely reworked to aware kernel version. See #10477. But, let's leave the issue for another PR.

Reply:
I agree with the kernel checking. The `IFLA_BOND_AD_INFO` is unrelated to the feature added since they are not nested into `IFLA_BOND_AD_INFO`. These are rather siblings in the netlink data.

Reply:
Another option could be to include a copy of the Kernel headers. This still would not handle the check of the kernel version but it may simplify the other parts. Having said that this could complicate things wrt. the rhel kernel.

Reply:
> But, let's leave the issue for another PR

Come on :-) Let's not break CentOS CI, which, by the way, has just started working thanks to https://github.com/systemd/systemd/pull/10589. I'd wait for https://bugs.centos.org/view.php?id=15386 to be resolved so that the CI could automatically verify that the code compiles there with no problems. Another option would be to ask @mrc0mmand to run the tests manually to make sure that it's safe to merge the PR.

Reply:
By the way, `gcc` seems to be complaining about `config_parse_ad_actor_system` on [SemaphoreCI](https://semaphoreci.com/systemd/systemd/branches/pull-request-10542/builds/4):
```
../src/network/netdev/bond.c: In function ‘config_parse_ad_actor_system’:
../src/network/netdev/bond.c:475:16: warning: missing braces around initializer [-Wmissing-braces]
         struct ether_addr null_addr = { 0 };
                ^
../src/network/netdev/bond.c:475:16: warning: (near initialization for ‘null_addr.ether_addr_octet’) [-Wmissing-braces]
```

Reply:
sorry, didn't have a look at the ci since it was not complaining. I'll have a look

Reply:
In general this should be fine and it could relate to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119 (not sure why this wasn't backproted to Ubuntu's 4.8.5 gcc)

But I could update this to {{ 0 }} to satisfy the old compiler.

Reply:
k, SemaphoreCI is happy now as well.

@evverx do you have an educated guess how long it will last until the CentOS CI ticket is resolved? I could give the build a try on a CentOS machine here, in case this last longer.

Reply:
Thanks for the link. To judge from the last comment there, it still isn't fixed in general: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80454. Anyway, I think `{ { 0 } }` should be fine and, as far as I can see, it won't be the first time it has been used in the systemd codebase:
https://github.com/systemd/systemd/blob/46054ac030c5aa498fcbe325fedec67dfc5b8134/src/systemd/sd-bus-vtable.h#L152-L157

Reply:
@toanju unfortunately, I don't know how long it will take to resolve that issue. FWIW, I tried to build the PR on a CentOS machine yesterday and it compiled. The only issue I ran into is `gcc` complaining about `{ 0 }` but it should be gone.

Reply:
Interestingly, `test_bond` has started to pass on CentOS. Apparently, the bonding options are supported there and I shouldn't have panicked :-)

Reply:
I am always amazed what is backported into the CentOS kernel. \o/

Reply:
@yuwata I restored the original state now wrt. meson.build, the missing.h should still be fine. Please have a look again. Thanks!

Reply:
Hmm, but `AD_INFO` was added in kernel-3.14...
https://github.com/torvalds/linux/commit/4ee7ac7526d4a9413cafa733d824edfe49fdcc46

Reply:
Ah sorry, I think now I get it... you want to check on the **last** attribute in that list.

Reply:
updated